storage/storage.go

```diff
@@ -46,8 +53,78 @@ type Handler interface {
 	RemoveSortedSetRange(string, string, string) error
 }
 
+const defaultHashAlgorithm = "murmur64"
+
+// If hashing algorithm is empty, use legacy key generation
+func GenerateToken(orgID, keyID, hashAlgorithm string) (string, error) {
+	if keyID == "" {
+		keyID = strings.Replace(uuid.NewV4().String(), "-", "", -1)
+	}
+
+	if hashAlgorithm != "" {
+		_, err := hashFunction(hashAlgorithm)
```

Comment:
I think it is a little bit of extra work here - we are allocating an instance of hash (which is runtime sys-call and takes extra memory for a bit) just to test if token needs to have a default hash algo, then we ignore allocated space which give some extra work for GC. I am not sure compiler can optimize this.

also I like more readable form `if _, err := hashFunction(hashAlgorithm); err != nil`

Reply:
also, we could introduce token as not just string but 1st class citizen in our code base as a struct, so we could put here in its fields hash algo name, Hasher func for later use in case we need to hash it (and do hash only once if it is not laready done), json and and base64 encoded  representation  of this token as fields as well